Lib/test/test_descr.py

```diff
@@ -1597,12 +1597,27 @@ class SubSpam(spam.spamlist): pass
         self.assertEqual(x2, SubSpam)
         self.assertEqual(a2, a1)
         self.assertEqual(d2, d1)
-        with self.assertRaises(TypeError):
+
+        with self.assertRaises(TypeError) as cm:
             spam_cm()
-        with self.assertRaises(TypeError):
+        self.assertEqual(
+            cm.exception.args[0],
```

Comment:
It may be better to test `str(cm.exception)` instead of `cm.exception.args[0]`. Just for the case if error messages be generated dynamically in the future.